packages/app/src/systems/Asset/components/AssetInput/AmountInput.tsx

```diff
@@ -0,0 +1,95 @@
+import { cssObj } from '@fuel-ui/css';
+import { Button, Flex, Input, Text } from '@fuel-ui/react';
+import type { BN } from 'fuels';
+import { useState } from 'react';
+import type { FC, ChangeEvent } from 'react';
+
+import { AmountInputLoader } from './AmountInputLoader';
+
+export type AmountInputProps = {
+  asset: {
```

Comment:
You can use this instead for the `asset` prop:

```ts
amounts: Coin[] | TxOutputCoin[] | TxInputCoin[];
```

Reply:
This is the error I get when I try to change the type to your suggestion.  I get the same error on all of the types.
```
Property 'format' does not exist on type 'Coin | CoinTransactionRequestOutput | CoinTransactionRequestInput'.
  Property 'format' does not exist on type 'Coin'.
```